networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/subnets/PeerSubnetSubscriptions.java

```diff
@@ -13,104 +13,157 @@
-    return IntStream.range(0, ATTESTATION_SUBNET_COUNT);
+  public int getSubscriberCountForSyncCommitteeSubnet(final int subnetId) {
+    return syncCommitteeSubnetSubscriptions.getSubscriberCountForSubnet(subnetId);
   }
 
-  @VisibleForTesting
-  static class Builder {
-    private final Map<Integer, Integer> subscriberCountBySubnetId = new HashMap<>();
+  public SszBitvector getAttestationSubnetSubscriptions(final NodeId peerId) {
+    return attestationSubnetSubscriptions.getSubnetSubscriptions(peerId);
+  }
 
-    private final Map<NodeId, SszBitvector> subscriptionsByPeer = new HashMap<>();
-    private int targetSubnetSubscriberCount = 2;
+  public SszBitvector getSyncCommitteeSubscriptions(final NodeId peerId) {
+    return syncCommitteeSubnetSubscriptions.getSubnetSubscriptions(peerId);
+  }
 
-    public Builder addSubscriber(final int subnetId, final NodeId peer) {
-      subscriberCountBySubnetId.put(
-          subnetId, subscriberCountBySubnetId.getOrDefault(subnetId, 0) + 1);
-      subscriptionsByPeer.compute(
-          peer,
-          (__, existingVector) ->
-              existingVector == null
-                  ? SUBNET_SUBSCRIPTIONS_SCHEMA.ofBits(subnetId)
-                  : existingVector.withBit(subnetId));
-      return this;
-    }
+  public boolean isSyncCommitteeSubnetRelevant(final int subnetId) {
+    return syncCommitteeSubnetSubscriptions.isSubnetRelevant(subnetId);
+  }
 
-    public Builder targetSubnetSubscriberCount(final int targetSubnetSubscriberCount) {
-      this.targetSubnetSubscriberCount = targetSubnetSubscriberCount;
-      return this;
-    }
+  public boolean isAttestationSubnetRelevant(final int subnetId) {
+    return attestationSubnetSubscriptions.isSubnetRelevant(subnetId);
+  }
 
-    public PeerSubnetSubscriptions build() {
-      return new PeerSubnetSubscriptions(
-          subscriberCountBySubnetId, subscriptionsByPeer, targetSubnetSubscriberCount);
+  public PeerScorer createScorer() {
+    return SubnetScorer.create(this);
+  }
+
+  public int getSubscribersRequired() {
+    return getMinSubscriberCount().map(c -> Math.max(targetSubnetSubscriberCount - c, 0)).orElse(0);
+  }
+
+  private Optional<Integer> getMinSubscriberCount() {
+    final Optional<Integer> minAttestationSubscribers = getMinAttestationSubscriberCount();
+    final Optional<Integer> minSyncnetSubscribers = getMinSyncCommitteeSubscriberCount();
+    if (minAttestationSubscribers.isPresent() && minSyncnetSubscribers.isPresent()) {
+      return Optional.of(Math.min(minAttestationSubscribers.get(), minSyncnetSubscribers.get()));
+    } else {
+      return minAttestationSubscribers.or(() -> minSyncnetSubscribers);
     }
   }
 
+  private Optional<Integer> getMinAttestationSubscriberCount() {
+    return attestationSubnetSubscriptions
+        .streamRelevantSubnets()
+        .map(attestationSubnetSubscriptions::getSubscriberCountForSubnet)
+        .min(Integer::compare);
+  }
```

Comment:
nit: Would it make sense to move this to be a method in `SubnetSubscriptions`?

Reply:
yep - good call 👍